LWC_Assignment copy/force-app/main/default/classes/CustomObjectController.cls

```diff
@@ -0,0 +1,75 @@
+public with sharing class CustomObjectController {
+    
+    @AuraEnabled(cacheable=true)
+    public static List<Company__c> fetchRecords() {
+        return [
+            SELECT Id,Name, Email__c
+            FROM Company__c
+            WITH SECURITY_ENFORCED
+        ]; 
+    }
+
+    @AuraEnabled
+    public static List<Company__c> searchCompanies(String searchTerm) {
+        
+        return [SELECT Id,Name, Email__c FROM Company__c WHERE Name LIKE :('%' + searchTerm + '%')];
```

Comment:
We need to add proper null checks on the apex side as this would throw an error if the list returns no rows